academic_validation_framework/database_integrations/base.py

```
@@ -0,0 +1,555 @@
+        """Cleanup database resources."""
+        if self._connection:
+            try:
+                await self._close_connection()
+            except Exception:
+                pass  # Ignore cleanup errors
+            finally:
+                self._connection = None
+    
+    async def _close_connection(self) -> None:
+        """Close database connection (implement in subclasses if needed)."""
+        pass
```

Ruff (0.11.9)

`BaseDatabaseIntegrationTester._close_connection` is an empty method in an abstract base class, but has no abstract decorator

(B027)